pkg/virt-controller/watch/migration.go

```diff
@@ -419,13 +419,6 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 		migrationCopy.Status.Phase = virtv1.MigrationFailed
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "VMI's migration state was cleared during the active migration.")
 		log.Log.Object(migration).Error("vmi migration state cleared during migration")
-	} else if migration.TargetIsHandedOff() &&
-		vmi.Status.MigrationState != nil &&
-		vmi.Status.MigrationState.MigrationUID != migration.UID {
-
-		migrationCopy.Status.Phase = virtv1.MigrationFailed
-		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "VMI's migration state was taken over by another migration job during active migration.")
-		log.Log.Object(migration).Error("vmi's migration state was taken over by another migration object")
 	} else if vmi.Status.MigrationState != nil &&
```

Comment:
is there any harm in leaving this check in? It seems like your fix should resolve issue where this gets triggered inaccurately.

Reply:
no harm, but it's essentially dead code.

Reply:
I would keep it so we can catch this kind of error in the future.

Reply:
Actually, I wouldn't, I've explained thoroughly in the commit message how this cannot happen as we have more than 1 layer of protection against this, that said I'd not keep it as this condition was the root cause of the bug that this PR fixes, basically by leaving this here we don't increase correctness we do the opposite by increasing the chance of false positives.

Reply:
I think this is going to only catch cases when something is really wrong as you fixed the root cause a couple of lines below.

Reply:
As requested by @xpivarc I re-introduced this check to be more conservative, even if I'm still for removing it, that said the newly added e2e test is still there.

Reply:
@acardace i agree that it seems impossible for this to happen. That said, if it did happen someday due to some unexpected change in behavior, i'd really like to know :D